CanDIG/clinical_ETL_code
# DIG-1671: Adding date format to manifest

### What's new
- Added a required date_format field to the manifest. It specifies the order of day, month and year to parse ambiguous dates properly.
- Updated test data and manifest with date formats.
- Improved mapping exceptions to be more readable and specific

### How to test
- Add date_format to the manifest
- Run CSVConvert.py using a dataset with ambiguous dates for that field and try different patterns in the template to confirm they are parsed as indicated.
- Test removing date resolutions and references to see the exceptions.

Previously:
~~Added a `date_format` parameter to the `date_interval` mapping function to specify the order of day, month and year to parse ambiguous dates properly.~~

~~How to test~~
~~- Add `date_format` to any date field in the template~~
~~DONOR.INDEX.date_of_birth, {date_interval(Donor.date_of_birth, date_format="DMY")}~~
~~- Run `CSVConvert.py` using a dataset with ambiguous dates for that field and try different patterns in the template to confirm they are parsed as indicated.~~
